rustls/src/msgs/base.rs

```diff
@@ -192,6 +192,43 @@ impl fmt::Debug for PayloadU8 {
     }
 }
 
+/// An arbitrary, unknown-content, non-empty, u8-length-prefixed payload
+#[derive(Clone, Eq, PartialEq)]
+pub(crate) struct PayloadNonZeroU8(pub(crate) Vec<u8>);
```

Comment:
Instead of duplicating, have you considered having a `const` generic argument for the minimum?

Reply:
Have tried this for the PayloadU16 type, will extend it to the U8 type if you think it's an improvement? I'm on the fence.

Reply:
Why are you on the fence?

Reply:
Hiding the mysterious type parameter via a type alias, which breaks the existing tuple struct uses.

I also tried this kind of thing:

```rust
pub trait Cardinality: Clone + Eq + PartialEq {
    const MIN: usize;
}

#[derive(Clone, Eq, PartialEq)]
pub struct MaybeEmpty;
impl Cardinality for MaybeEmpty {
    const MIN: usize = 0;
}

#[derive(Clone, Eq, PartialEq)]
pub struct NonEmpty;
impl Cardinality for NonEmpty {
    const MIN: usize = 1;
}
```

and then have `PayloadU16<C: Cardinality = MaybeEmpty>` for existing code, and `PayloadU16<NonEmpty>` for the changes in this PR. That also breaks the existing tuple struct uses (because the need for a `PhantomData<C>`) but I suppose that is not a huge cost.

Reply:
I'd just get rid of the type aliases, which I don't like anyway (except perhaps as a transition mechanism). In general avoiding the duplication in favor of abstraction seems like a good thing?